src/BloomExe/web/ServerBase.cs

```diff
@@ -246,52 +211,125 @@ protected static string GetLocalPathWithoutQuery(IRequestInfo info)
 			}
+		}
 
-			if (requestedPath.EndsWith("testconnection"))
+		/// <summary>
+		/// TODO: Note: doing this at runtime isn't as good as doing it in the installer, because we have no way of
+		/// removing these entries on uninstall (but the installer does).
+		/// </summary>
+		private static void AddUrlAccessControlEntry()
+		{
+			MessageBox.Show(
+				string.Format("We need to do one more thing before Bloom is ready. Bloom needs temporary administrator privileges to set up part of its communication with the embedded web browser.{0}{0}After you click 'OK', you may be asked to authorize this step.", Environment.NewLine),
+					@"Almost there!", MessageBoxButtons.OK);
+
+			var startInfo = new ProcessStartInfo { UseShellExecute = true, Verb = "runas" };
+
+			if (Environment.OSVersion.Version.Major == 5 /*win xp*/)
 			{
-				info.WriteCompleteOutput("OK");
-				return true;
+				startInfo.FileName = "httpcfg";
+				startInfo.Arguments = string.Format("set urlacl -u {0} /a \"D:(A;;GX;;;WD)\"", PathEndingInSlash);
 			}
-			return false;
+			else
+			{
+				startInfo.FileName = "netsh";
+				startInfo.Arguments = string.Format("http add urlacl url={0} user=everyone", PathEndingInSlash.TrimEnd('/'));
+			}
+			Process.Start(startInfo);
 		}
 
-		public static string PathEndingInSlash
+		private static bool GetIsAbleToUsePort()
 		{
-			get { return "http://localhost:8089/bloom/"; }
+			try
+			{
+				var x = new WebClientWithTimeout { Timeout = 3000 };
+				return ("OK" == x.DownloadString(PathEndingInSlash + "testconnection"));
+			}
+			catch (Exception e)
+			{
+				return false;
+			}
 		}
 
-		public static string GetContentType(string extension)
+		#region Disposable stuff
+
+		private bool IsDisposed { get; set; }
+
+		public void Dispose()
 		{
-			switch (extension)
+			//Stop();
+			Dispose(true);
+			GC.SuppressFinalize(this);
+		}
+
+		protected virtual void Dispose(bool fDisposing)
+		{
+			Debug.WriteLineIf(!fDisposing, "****** Missing Dispose() call for " + GetType() + ". *******");
+			if (fDisposing && !IsDisposed)
 			{
-				case ".css":  return "text/css";
-				case ".gif":  return "image/gif";
-				case ".htm":
-				case ".html": return "text/html";
-				case ".jpg":
-				case ".jpeg": return "image/jpeg";
-				case ".js":   return "application/x-javascript";
-				case ".png":  return "image/png";
-				case ".pdf":  return "application/pdf";
-				case ".txt":  return "text/plain";
-				case ".svg":  return "image/svg+xml";
-				default:      return "application/octet-stream";
+				// dispose managed and unmanaged objects
+				try
+				{
+					_isDisposing = true;
+
+					if (_listener != null)
+					{
+						//prompted by the mysterious BL 273, Crash while closing down the imageserver
+						Guard.AgainstNull(_listenerThread, "_listenerThread");
+						//prompted by the mysterious BL 273, Crash while closing down the imageserver
+						Guard.AgainstNull(_stop, "_stop");
+
+						_stop.Set();
```

Comment:
Is one Set() enough to make all five threads stop what they are doing? Note also my comment above about whether it will really stop a worker thread if the queue is empty.